zoloto/cameras/camera.py

```diff
@@ -49,6 +49,13 @@ def discover(cls, **kwargs: Any) -> Generator["Camera", None, None]:
         for camera_id in find_camera_ids():
             yield cls(camera_id, **kwargs)
 
+    def __iter__(self) -> Generator[ndarray, None, None]:
```

Comment:
The code for this method is identical between the three `Camera`s that have it.

I'd prefer if you used an `IterableCameraMixin`, or similar.

e.g

```python
class IterableCameraMixin(ABC):

   def __iter__(self) -> Generator[ndarray, None, None]:
        while True:
            frame = self.capture_frame()
            if not frame.size:
                break
            yield frame

class Camera(BaseCamera, IterableCameraMixin):
    ...
```

Reply:
I too was thinking this, I forget why I never went for it. Will give it a play